cmd/lotus-storage-miner/init_restore.go

```diff
@@ -30,250 +31,260 @@ import (
 	"github.com/filecoin-project/lotus/node/repo"
 )
 
-var initRestoreCmd = &cli.Command{
-	Name:  "restore",
-	Usage: "Initialize a lotus miner repo from a backup",
-	Flags: []cli.Flag{
-		&cli.BoolFlag{
-			Name:  "nosync",
-			Usage: "don't check full-node sync status",
-		},
-		&cli.StringFlag{
-			Name:  "config",
-			Usage: "config file (config.toml)",
-		},
-		&cli.StringFlag{
-			Name:  "storage-config",
-			Usage: "storage paths config (storage.json)",
-		},
-	},
-	ArgsUsage: "[backupFile]",
-	Action: func(cctx *cli.Context) error {
-		log.Info("Initializing lotus miner using a backup")
-		if cctx.Args().Len() != 1 {
-			return xerrors.Errorf("expected 1 argument")
-		}
+// ctx context.Context, api lapi.FullNode, addr address.Address, peerid peer.ID
+func restore(ctx context.Context, cctx *cli.Context, manageConfig func(*config.StorageMiner) error, after func(api lapi.FullNode, addr address.Address, peerid peer.ID, mi miner.MinerInfo) error) error {
+	if cctx.Args().Len() != 1 {
+		return xerrors.Errorf("expected 1 argument")
+	}
 
-		ctx := lcli.ReqContext(cctx)
+	log.Info("Trying to connect to full node RPC")
 
-		log.Info("Trying to connect to full node RPC")
+	api, closer, err := lcli.GetFullNodeAPIV1(cctx) // TODO: consider storing full node address in config
+	if err != nil {
+		return err
+	}
+	defer closer()
 
-		if err := checkV1ApiSupport(ctx, cctx); err != nil {
-			return err
-		}
+	log.Info("Checking full node version")
 
-		api, closer, err := lcli.GetFullNodeAPIV1(cctx) // TODO: consider storing full node address in config
-		if err != nil {
-			return err
-		}
-		defer closer()
+	v, err := api.Version(ctx)
+	if err != nil {
+		return err
+	}
 
-		log.Info("Checking full node version")
+	if !v.APIVersion.EqMajorMinor(lapi.FullAPIVersion1) {
+		return xerrors.Errorf("Remote API version didn't match (expected %s, remote %s)", lapi.FullAPIVersion1, v.APIVersion)
+	}
 
-		v, err := api.Version(ctx)
+	if !cctx.Bool("nosync") {
+		if err := lcli.SyncWait(ctx, &v0api.WrapperV1Full{FullNode: api}, false); err != nil {
```

Comment:
Suggest adding a log line here as sync wait can probably take a while

Reply:
When we don't have `nosync`, we output the result of `lotus sync wait` and it is clear that the miner is waiting for sync.